x-pack/plugins/fleet/public/applications/integrations/app.tsx

```diff
@@ -73,17 +74,20 @@ export const IntegrationsAppContext: React.FC<{
                       <UIExtensionsContext.Provider value={extensions}>
                         <FleetStatusProvider>
                           <startServices.customIntegrations.ContextProvider>
-                            <Router history={history}>
-                              <AgentPolicyContextProvider>
-                                <PackageInstallProvider
-                                  notifications={startServices.notifications}
-                                  theme$={theme$}
-                                >
-                                  <IntegrationsHeader {...{ setHeaderActionMenu, theme$ }} />
-                                  {children}
-                                </PackageInstallProvider>
-                              </AgentPolicyContextProvider>
-                            </Router>
+                            <startServices.engagement.ContextProvider>
+                              <Router history={history}>
+                                <AgentPolicyContextProvider>
+                                  <PackageInstallProvider
+                                    notifications={startServices.notifications}
+                                    theme$={theme$}
+                                  >
+                                    <IntegrationsHeader {...{ setHeaderActionMenu, theme$ }} />
+                                    {children}
+                                    <EngagementChat />
```

Comment:
Is this `engagement.ContextProvider` only used for the `EngagementChat` component? If so, did you though about exposing a context-wrapper component via the `engagement` plugin's contract instead of having to use both the plugin's contract to access the context provider and direct imports to import the chat component?


something like 


```ts
<Router history={history}>
  <....>
    <startServices.engagement.Chat />
  </...>
</Router>
```

this is a pattern commonly used in other parts of Kibana when in need of exposing preconfigured/context-injected components to other plugins

Reply:
The idea was for this plugin to have more than one component, so the initial API matters.

That's an interesting approach!  I'll have to think about it a bit more.  My concerns with it at the moment are:
- getting `startServices` down the tree in cases where a component is more granular, (e.g. a configured drop down control.
- storing *all* `startServices` in context in the off chance you use a single component, whereas the `ContextProvider` simply provides services for components.  I'm not sure if the difference is meaningful or not.
- mocking the `startServices` in Storybook and Jest tests has been notoriously difficult.  I'm loathe to introduce a complex `startServices` construct when you're only relying on `ContextProvider`.

I need to look into this more.